x-pack/filebeat/module/rabbitmq/log/ingest/pipeline.json

```diff
@@ -0,0 +1,37 @@
+{
+  "description": "Pipeline for parsing RabbitMQ logs",
+  "processors": [
+    {
+      "grok" : {
+        "field" : "message",
+        "pattern_definitions" : {
+          "GREEDYMULTILINE" : "(.|\n)*",
+          "ERL_PID" : "\\<%{INT}+\\.%{INT}+\\.%{INT}+\\>"
+        },
+        "patterns" : [
+          "%{DATESTAMP:timestamp} \\[%{WORD:log.level}\\] %{ERL_PID:rabbitmq.log.pid} %{GREEDYMULTILINE:message}"
+        ],
+        "ignore_missing" : true
+      }
+    },
+    {
+      "date" : {
+        "field" : "timestamp",
+        "formats" : ["yy-MM-dd HH:mm:ss.SSS"],
+        {< if .convert_timezone >}"timezone": "{{ event.timezone }}",{< end >}
```

Comment:
I copied this from other modules but have no idea how `{< if .convert_timezone >}` works. Are there any docs about this syntax?

I assume it's a know issue/tradeoff but this makes the file invalid as JSON.

Reply:
Yes, it's a know tradeoff as it's the way that makes it possible for us to have pipeline templates. Recently a few modules started to use yaml instead of json but I assume the above template language will also make it invalid yaml. @adriansr might know?

Reply:
Pipelines and configuration files for a module are preprocessed with [Golang template package](https://golang.org/pkg/text/template/) (Using `{< >}` instead of `{{ }}`as delimiters).

I didn't had a problem with the files being valid JSON/YAML because my editor doesn't complain too much. But if this is a problem, you can switch the pipeline to YAML and hide the actions in a comment:

```
- date:
     field: "_temp_.generated_time"
     ignore_failure: true
     #{< if .convert_timezone >}
     timezone: "{{ event.timezone }}"
     #{< end >}
     formats:
       - "yyyy/MM/dd HH:mm:ss"
```

Reply:
Nice trick with the commends in yaml. I like it.